### PR TITLE
feat: add nvidia driver support

### PR DESCRIPTION
**Pull request by mhrabovcin**

### CODE REVIEW

File: ansible/group_vars/all/system.yaml [hunk 1]
```diff
@@ -13,13 +13,22 @@ docker_rpm_repository_url_upstream_mirror: "https://packages.d2iq.com/download.d
...
 nvidia_container_runtime_gpg_key_url: "https://nvidia.github.io/nvidia-container-runtime/gpgkey"
+nvidia_driver_rpm_repository_package_url: http://developer.download.nvidia.com/compute/cuda/repos/rhel{{ ansible_distribution_major_version|int }}/x86_64/
+nvidia_driver_rpm_gpg_key_url: http://developer.download.nvidia.com/compute/cuda/repos/rhel{{ ansible_distribution_major_version|int }}/x86_64/7fa2af80.pub
+# On Centos7 installs `nvidia-driver` with dkms for kernel line 3.x w version
+nvidia_driver_package_name: "nvidia-driver-latest-dkms-3"
+nvidia_driver_package_version: "460.73.01-1.el7.x86_64"
+# On Centos8 installs `nvidia-driver` module with given version
+nvidia_driver_module_name: "nvidia-driver"
+nvidia_driver_module_version: "460-dkms"
+nvidia_driver_deb_repository_url: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/
```
Comment (jongiddy) [inline ansible/group_vars/all/system.yaml:31]: It's not immediately clear what the difference here is between `package` and `module`. Can we perhaps expand the names to `...rpm_package...` and `...dnf_module...`? That then matches the `...deb_repository...` pattern.

File: ansible/roles/packages/tasks/nvidia.yaml [hunk 2]
```diff
@@ -7,4 +7,79 @@
...
+        - ansible_os_family == 'RedHat'
+        - ansible_distribution == 'RedHat'
+        - ansible_distribution_major_version|int == 7
+    - name: Install NVIDIA driver (centos/rhel 8)
+      dnf:
+        name: >-
+          @{{ nvidia_driver_module_name }}:{{ nvidia_driver_module_version }}
+        state: present
+      when:
+        - ansible_os_family == 'RedHat'
```
[the remark this answers is not in the document]
Comment (mhrabovcin) [reply ansible/roles/packages/tasks/nvidia.yaml:84]: I'll remove the extra check from rhel7.

File: ansible/group_vars/all/system.yaml (hunk 1 above)
Comment (mhrabovcin) [reply ansible/group_vars/all/system.yaml]: Fixed.

**Review by jongiddy — APPROVED** [review]
:+1:

File: docs/nvidia-gpu.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    --override overrides/nvidia.yaml
+```
+
+## Supported images
+
+| base os | nvidia             |
+|---------|--------------------|
+| centos7 | :white_check_mark: |
+| centos8 | :white_check_mark: |
+| flatcar | :x:                |
```
Comment (tillt) [inline docs/nvidia-gpu.md:30]: Great, thanks for this. We can re-use this later, when konvoy-image-builder comes with documentation on its own.

**Review by tillt — APPROVED** [review]

**Pull request merged**